### DISCUSSION TITLE
[v8] How can I have a custom filter component for each column?

### DISCUSSION THREAD

**Discussion by bertuccellimatteo** [TanStack/table#4314] [Q&A]
In react-table v7 I could use a custom filter component for each column by declaring it in the column definition like this:
`const columns = [{ accessor: "name", Filter: MyFilterComponent }]`
In the v8 column def api however you cannot do it anymore, I tried a workaround using column meta that works but does not seem the cleanest way to handle this, do you have any suggestions?

**Comment by CalMlynarczyk** [comment]
[The library creator has stated that he does not see a way to add this back in v8](https://github.com/TanStack/table/discussions/4018#discussioncomment-5851201) (I do not know why this is a problem with v8 but not v7), so your only option for now is using `meta` on the column definition. [There is an answer on that post about how to implement this](https://github.com/TanStack/table/discussions/4018#discussioncomment-3988293).
